recursive_art.py

```diff
@@ -17,12 +22,50 @@ def build_random_function(min_depth, max_depth):
 
     Returns:
         The randomly generated function represented as a nested list.
-        (See the assignment writ-eup for details on the representation of
+        (See the assignment write-up for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
-
+    #TODO: make depth random
+    if max_depth == 0:
+        xOry = random.randint(1,2)
+        if xOry == 1:
+            return ['x']
+        else:
+            return ['y']
+    if (min_depth ==0) and (random.randint(1,2)==1):
+        xOry = random.randint(1,2)
+        if xOry == 1:
+            return ['x']
+        else:
+            return ['y']
+    else:
+        myInt = random.randint(1,4)
+        if myInt == 1:
+            l1 = ['prod']
+            l2 = (build_random_function(min_depth-1, max_depth-1))
+            l3 = (build_random_function(min_depth-1, max_depth-1))
+            l1.append(l2)
+            l1.append(l3)
+            return l1
+        elif myInt == 2:
+            l1 = ['avg']
+            l2 = (build_random_function(min_depth-1, max_depth-1))
+            l3 = (build_random_function(min_depth-1, max_depth-1))
+            l1.append(l2)
+            l1.append(l3)
+            return l1
+        elif myInt == 3:
+            l1 =  ['cos_pi']
+            l2 = (build_random_function(min_depth-1, max_depth-1))
+            l1.append(l2)
+            return l1
+        elif myInt == 4:
+            l1 =  ['sin_pi']
+            l2 = (build_random_function(min_depth-1, max_depth-1))
+            l1.append(l2)
+            return l1
+
+print("ran func: " + (str)(build_random_function(3,4)))
```

Comment:
- You can remove to TODO comment at the top
- If you use `l = lambda : build_random_function(min_depth-1, max_depth-1)`, and use `l()` whenever you need them, you don't need to repeat l2, l3 codes and write out similar code for each if statement.
for example, the code would look like
```
elif myInt == 2:
    l1 = ['avg']
    l1.append(l())
    l1.append(l())
    return l1
```